### PR TITLE
[dtemplate] use early return

### PR DESCRIPTION
**Pull request by thewilsonator**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

**Comment by TungstenHeart** [comment]
I just wonder why have you forced pushed to [add an emtpy line](https://github.com/dlang/dmd/compare/852cf8bc25a6c07f00c4b8e895c2c0c7e9b7d5f8..4247297ca3c09058b57ab2ce87fef0210be78881) ? CI issues ?

**Comment by thewilsonator** [comment]
> CI issues ?

The CI complains about trailing whitespace which I despise.

File: src/dmd/dtemplate.d [hunk 1]
```diff
@@ -2697,104 +2697,104 @@ void functionResolve(ref MatchAccumulator m, Dsymbol dstart, Loc loc, Scope* sc,
...
+    LfIsBetter:
+        td_best = null;
+        ti_best = null;
+        ta_last = MATCH.exact;
+        m.last = mfa;
+        m.lastf = fd;
+        tthis_best = tthis_fd;
+        ov_index = 0;
+        m.count = 1;
+        return 0;
```
Comment (ibuclaw) [inline src/dmd/dtemplate.d:2796]: Make these nested functions?

Comment (12345swordy) [reply src/dmd/dtemplate.d:2796]: I concur, lets avoid using gotos as much as possible.

**Pull request merged**